src/button.scss

```diff
@@ -11,456 +45,454 @@
+    font-size: $fd-menu-button-icon-size;
+    position: absolute;
+    top: $fd-menu-button-icon-position-top;
   }
 
-  @include fd-active-pressed-selected() {
-    color: var(--sapButton_Selected_TextColor);
-    background-color: var(--sapButton_Accept_Active_Background);
-    border-color: var(--sapButton_Accept_Active_BorderColor);
-
-    &:focus {
-      outline-color: var(--sapContent_ContrastFocusColor);
+  &:empty {
+    &::before {
+      margin-right: 0;
+      margin-left: 0;
     }
 
-    @include fd-disabled() {
-      @include isDisabled();
-      @include positive();
+    &::after {
+      margin-right: $fd-button-icon-margin;
+      margin-left: $fd-button-icon-margin;
+    }
+  }
 
-      box-shadow: none;
+  @at-root {
+    [dir="rtl"] &,
+    &[dir="rtl"] {
+      padding-left: $fd-button-menu-button-padding;
+      padding-right: $fd-button-padding-x;
 
-      &:focus {
-        outline: none;
+      &:empty {
+        &::before {
+          margin-left: 0;
+        }
       }
     }
   }
+}
 
-  &:focus {
-    @include focus();
+@mixin textAlignment() {
+  &-left {
+    text-align: left;
   }
-}
 
-@mixin isLight() {
-  border-color: var(--sapButton_Lite_BorderColor);
-  background-color: var(--sapButton_Lite_Background);
-}
+  &-right {
+    text-align: right;
+  }
 
-@mixin isEmphasized() {
-  color: var(--sapButton_Emphasized_TextColor);
-  border-color: var(--sapButton_Emphasized_BorderColor);
-  background-color: var(--sapButton_Emphasized_Background);
+  &-center {
+    text-align: center;
+  }
 }
 
-/*!
-.fd-button+( (--compact), (--emphasized, --light), (--standard, --positive, --medium, --negative))+( (.is-disabled | [aria-disabled=true]) | (.is-selected | [aria-selected=true] | (.is-pressed | [aria-pressed=true]))
-*/
-
-$block: #{$fd-namespace}-button;
-
-.#{$block},
-a.#{$block} {
-  @include isStandard();
-  // spacing
-  $fd-button-padding-x: fd-space("xs") !default;
-  // anim
-  $fd-button-transition-params: $fd-animation--speed ease-in !default;
+@mixin half() {
+  --fd-forms-height-half: #{$fd-forms-height/2};
 
-  // BASE
-  // set all reset and baseline block styles
-  @at-root {
-    .#{$block},
-    [class*="#{$block}--"],
-    a[class*="#{$block}--"] {
-      @include fd-reset();
-      @include fd-button-reset();
-      @include isStandard();
-      // set metrics
-      @include fd-var-size("height", $fd-forms-height, --fd-forms-height);
-      @include fd-var-size("max-height", $fd-forms-height, --fd-forms-height);
-      @include fd-var-size("min-width", $fd-forms-height, --fd-forms-height);
-      @include fd-type("0");
-      @include fd-icon-size("m", "before");
-      @include fd-icon-size("m", "after");
-      @include fd-var-size("line-height", $fd-button-line-height, --fd-button-line-height);
-
-      padding-left: $fd-button-padding-x;
-      padding-right: $fd-button-padding-x;
-      text-align: center;
+  @include fd-var-size("height", $fd-forms-height/2, --fd-forms-height-half);
```

Comment:
Shouldn't we get rid of values based on fundamental styles values?

Reply:
fixed

Reply:
I think we shouldn't use these. They are still defined in px

Reply:
fixed ;)